### DISCUSSION TITLE
Reconstruction issues using MAP-MRI  model (RTOP, RTAP, RTPP)

### DISCUSSION THREAD

**Discussion by CATHYCW** [dipy/dipy#2855] [Scientific Visualization] [answered]
## Description
We are doing a project about Intermittent Exotropia, and we need to reconstruct the brain parameters using MAP-MRI model

When we run the example code reconst_mapmri.ipynb [,](https://dipy.org/documentation/1.7.0/examples_built/07_reconstruction/reconst_mapmri/#sphx-glr-examples-built-07-reconstruction-reconst-mapmri-py) it is good when we use the image fragment data_small [40:65, 50:51]. However, if the selected image fragments including the area outside the brain, the image of RTOP, RTAP, and RTPP parameter will be wrong, which contain no signals. The correct and wrong outputs are shown below.

Wrong output (contain the area outside brain):
![image](https://github.com/dipy/dipy/assets/138023015/36d2e0b8-53d5-402f-825f-51b44420b98f)
Correct output (only contain the area inside the brain):
![image](https://github.com/dipy/dipy/assets/138023015/3076ffea-3791-4fc2-a11d-a77197853b1f)

We are trying to use a mask but we are wondering if there exists other solutions. Thanks a lot!

**Comment by CATHYCW** [comment]
We have tried the preprocessing step to add a mask by [Brain segmentation with median_otsu](https://dipy.org/documentation/1.7.0/examples_built/04_preprocessing/brain_extraction_dwi/), but it does not work.

**Comment by arokem** [comment]
I believe that this may be only a visualization issue. When you expand the field of view, your volume includes some voxels that contain very extreme values (this happens because of aphysical signals, e.g., at the boundary between brain and skull). If you visualize the data from the whole brain but use a boundary on the range of values to display (e.g., the `vmin` and `vmax` key-word arguments to `matshow`), do you still see the same issue?

**Comment by CATHYCW** [comment]
Thanks for your comments. However, we save the output images and use ITK-SNAP (an visualization app) to check that. The problem still exists. Maybe it's not a visualization problem.

**Comment by pjsjongsung** [comment]
I'm not sure about what ITK snap does to remove extreme values during visualization. Can you try checking if the image intensity is also not as desired? Quick way to look at it would be using `dipy_horizon name_of_file.nii.gz` in the command line and clicking on the positions you want to check, which will show you the values of that position.

Also, could you elaborate more on median_otsu not working? Maybe we could help you with that as well.

**Comment by CATHYCW** [comment, marked as answer]
Thanks a lot for your reply. I check what you mentioned and find that the image we saved is 32-bit or greater precision. However, the visualization app can only show images with 16-bit precision. The warning is shown below.
![image](https://github.com/dipy/dipy/assets/138023015/1eea3d5a-241b-4670-98b9-26482f10dc5d)
We have tried to saved the image with int16 and shrinked the pixel value by 100 to make the outputs easier to visualization. However, this method seems not that reasonable. Therefore, we are wondering if there exists better solutions. Moreover, the median_otsu works well.

**Comment by pjsjongsung** [comment]
We have a viewer called horizon as mentioned in the previous comment. You can try using it by
`dipy_horizon file_path`
it doesn't have that much interface as ITK snap but for visualization purposes it should serve you well.
